File: frontend/projects/admin/src/app/dto/client.dto.ts
export interface ClientDTO {
  id?: number;
  nom: string;
  prenom: string;
  telephone: string;
  email?: string;
  adresse?: string;
  ville?: string;
  region?: string;
  dateNaissance?: string; // ISO date string
  sexe?: string;
  professionnel: boolean;
  entreprise?: string;
  secteurActif?: string;
  isActive?: boolean;
  createdAt?: string;
  updatedAt?: string;
}

File: backend/src/main/java/com/vollailelink/backend/dto/ClientDTO.java
package com.vollailelink.backend.dto;

import com.vollailelink.backend.model.enums.Sexe;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ClientDTO {
    private Long id;

    @NotBlank(message = "Le nom est obligatoire")
    private String nom;

    @NotBlank(message = "Le prénom est obligatoire")
    private String prenom;

    @NotBlank(message = "Le téléphone est obligatoire")
    @Pattern(regexp = "^\\+\\d{1,3}\\d{9,15}$", message = "Format de téléphone invalide (E.164 requis: +221...)")
    private String telephone;

    private String email;
    private String adresse;
    private String ville;
    private String region;
    private LocalDate dateNaissance;
    private Sexe sexe;
    private Boolean professionnel;
    private String entreprise;
    private String secteurActivite;
    private Boolean isActive;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}
